Userland/Libraries/LibGfx/ICCProfile.cpp

```diff
@@ -1019,13 +1019,213 @@ ErrorOr<void> Profile::read_tag_table(ReadonlyBytes bytes)
     return {};
 }
 
+static bool is_xCLR(ColorSpace color_space)
+{
+    switch (color_space) {
+    case ColorSpace::TwoColor:
+    case ColorSpace::ThreeColor:
+    case ColorSpace::FourColor:
+    case ColorSpace::FiveColor:
+    case ColorSpace::SixColor:
+    case ColorSpace::SevenColor:
+    case ColorSpace::EightColor:
+    case ColorSpace::NineColor:
+    case ColorSpace::TenColor:
+    case ColorSpace::ElevenColor:
+    case ColorSpace::TwelveColor:
+    case ColorSpace::ThirteenColor:
+    case ColorSpace::FourteenColor:
+    case ColorSpace::FifteenColor:
+        return true;
+    default:
+        return false;
+    }
+}
+
+ErrorOr<void> Profile::check_required_tags()
+{
+    // ICC v4, 8 Required tags
+
+    // ICC v4, 8.2 Common requirements
+    // "With the exception of DeviceLink profiles, all profiles shall contain the following tags:
+    //  - profileDescriptionTag (see 9.2.41);
+    //  - copyrightTag (see 9.2.21);
+    //  - mediaWhitePointTag (see 9.2.34);
+    //  - chromaticAdaptationTag, when the measurement data used to calculate the profile was specified for an
+    //    adopted white with a chromaticity different from that of the PCS adopted white (see 9.2.15).
+    //  NOTE A DeviceLink profile is not required to have either a mediaWhitePointTag or a chromaticAdaptationTag."
+    // profileDescriptionTag, copyrightTag are required for DeviceLink too (see ICC v4, 8.6 DeviceLink profile).
+    // profileDescriptionTag, copyrightTag, mediaWhitePointTag are required in ICC v2 as well.
+    // chromaticAdaptationTag isn't required in v2 profiles as far as I can tell.
+
+    if (!m_tag_table.contains(profileDescriptionTag))
```

Comment:
I realize these names match the spec but `camelCase` in serenity code looks very wrong - consider renaming these at some point

Reply:
Happy to do this at some point :P

They are declared here: https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibGfx/ICCProfile.h#L77 (and here: https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibGfx/ICCProfile.h#L117), also used here: https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibGfx/ICCProfile.cpp#L399

So the options are:
* Keep things as-is. Variable names look like in the spec, but unlike other symbols.
* Use upper-case name in the xmacro. Variable names look like expected, but `icc` output now prints tags uppercase instead of like in the spec.
* Add both lower-case and upper-case spellings to the xmacro

Happy to do whichever you like best (…but maybe not in this PR here).

Reply:
(IIRC some LibWeb symbols also match spec instead of serenity? But I might be making that up.)

Reply:
The only thing that comes to mind are LibJS' "common property names", where the identifier casing matches the underlying string value: https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibJS/Runtime/CommonPropertyNames.h
Definitely not for this PR & needs more opinions than just mine, just wanted to point it out as something that seemed off :smile: